e2e/__tests__/__snapshots__/failures.test.ts.snap

```diff
@@ -387,9 +387,9 @@ FAIL __tests__/snapshotNamed.test.js
 
   ● failing named snapshot
 
-    expect(value).toMatchSnapshot()
+    expect(received).toMatchSnapshot(name)
 
-    Received value does not match stored snapshot "failing named snapshot: snapname 1".
+    Snapshot key: \`failing named snapshot: snapname 1\`
```

Comment:
I don't think "snapshot key" is a good word for it. What about "snapshots name"?

Reply:
I thought the same, but on the other hand we have a "name" argument in a snapshot and it feels confusing as well :D

Reply:
Indeed I used incorrect analogy from properties of object which have **keys** instead of bindings of module which have **names**.

A goal of the formatting change is so tester can copy value from console to paste into Find box of editor (without wrong escaping for a double quoted string in edge cases).

Let’s decide on terms for the report which will also be clear when I add to docs an explicit  relationship between name arguments of enclosing blocks and optional `whatever` argument of matcher to the export-by-name from `.snap` file.

Given picture with silly placeholder for optional arg, what do y’all suggest to move forward:

<img width="600" alt="toMatchSnapshot improved 2" src="https://user-images.githubusercontent.com/11862657/54493335-29d2df00-48a5-11e9-807b-f8305ff03ba3.png">

Reply:
Oh, what do y’all think about `suffix` for the optional part of the name?

Reply:
Hm, or how about "snapshot title" and "subtitle" instead of suffix? Still not sure about it

Reply:
How about bold weight for the optional whatever-it-is substring in the template literal?

For the matcher hint:
* Could just omit mention of the whatever arg
* If (I think it is) better to display the arg in the hint, but not as a separate labelled line, I researched sequence of issues and pull requests how we got here, the goal is distinguish the purpose or meaning of multiple snapshots per `it` or `test` block. None of my suggestions have communicated that. A vague try is `comment` but you can do better :)

P.S. Any thoughts how to tactfully explain costs of multiple assertions per test are welcome for future update to docs. For example, when an assertion fails, Jest stops testing any other assertions in the `it` or `test` block.

Reply:
I think bold makes sense. Also agreed on the P.S., people should be aware of that.

Reply:
What do you think about `hint` as analogy to password hint:

> A reminder of how a password was derived. In order to jog the user's memory, some login systems allow a hint to be entered, which is displayed each time the password is requested.

<img width="600" alt="toMatchSnapshot properties improved 3" src="https://user-images.githubusercontent.com/11862657/54540181-e5f7dc80-496d-11e9-919a-1cb041e55415.png">

The optional `name` argument:
* was **instead of** test name in #2094
* but was not renamed (pardon pun :) when it became **in addition to** test name in #4460
* which was more like the original request in #1392

If you agree with this change:
* I will update `ExpectAPI.md` in this PR to rename the argument as `snapshotHint`
* I would like to keep `Snapshot name:` as label for the entire string as analogy to named exports from `.snap` files, and open a future docs PR to explain it in `SnapshotTesting.md`

Reply:
Hint sounds good. Can you also make the "hint" in `matcherHint` bold so it corresponds nicely with what's inside "snapshot name"?

Reply:
Yes, your wish is my command. Bold green is a subtle distinction, in a good way, I think:

<img width="600" alt="toMatchSnapshot properties improved 4" src="https://user-images.githubusercontent.com/11862657/54548168-b2bd4980-497d-11e9-8346-3967dc8452f0.png">